controller/backup_controller.go

```diff
@@ -813,17 +824,28 @@ func (bc *BackupController) checkMonitor(backup *longhorn.Backup, volume *longho
 	if engine.Status.CurrentState != longhorn.InstanceStateRunning ||
 		engine.Spec.DesireState != longhorn.InstanceStateRunning ||
 		volume.Status.State != longhorn.VolumeStateAttached {
+		bc.creationRetryCounter.IncreaseCount(backup.Name)
+		if bc.creationRetryCounter.GetCount(backup.Name) >= maxCreationRetry {
```

Comment:
It would have simplified things a little if IncreaseCount just returned the current count, which can be checked.  Generally don't need the extra lock, map traversal, and unlock to turn right around and get it after increment.

Reply:
Good suggestion. But PR is already merged so I will keep it as it is for now. Is it ok?

Reply:
Sure.  We can improve it if we have reason to touch it again.